### PR TITLE
[SPARK-11515][ML] QuantileDiscretizer should take random seed

### PR DESCRIPTION
**Pull request by yu-iskw**
cc @jkbradley

### CODE REVIEW

**Comment by yu-iskw** [comment]
Jenkins, test this please.

**Comment by yu-iskw** [comment]
I have rebased this PR with master.

**Comment by thunterdb** [comment]
@yu-iskw thanks for fixing this issue. It looks good to me, but can you please resolve the conflicts?

**Comment by jkbradley** [comment]
@yu-iskw Apologies for the delay.  We can get this merged now, once you can update it.  Thank you!

File: mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala [hunk 1]
```diff
@@ -23,8 +23,8 @@ import org.apache.spark.Logging
 import org.apache.spark.annotation.{Experimental, Since}
 import org.apache.spark.ml._
 import org.apache.spark.ml.attribute.NominalAttribute
+import org.apache.spark.ml.param.shared.{HasSeed, HasInputCol, HasOutputCol}
```
Comment (thunterdb) [inline mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala]: @yu-iskw you need to respect the lexicographic order for the imports

Comment (yu-iskw) [inline mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala]: Thank you for pointing out. I forgot running the lint script.

**Comment by yu-iskw** [comment]
@jkbradley I have rebased with master. Can you review it?

**Comment by yu-iskw** [comment]
@jkbradley ping

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks! @yu-iskw I created https://issues.apache.org/jira/browse/SPARK-13292 for the Python API and assigned to you.

**Comment by yu-iskw** [comment]
Thank you for merging it. I will do the issue. Thanks!

**Pull request closed without merging**